src/noitedit/read.c

```diff
@@ -271,8 +271,6 @@ read_char(EditLine *el, char *cp)
 		else
 			num_read = read(el->el_infd, cp, 1);
 
-		if (num_read)
-			break;
```

Comment:
This is unlikely to be correct.  I think it should say and be changed to `if(num_read > 0) break;`